python/railjson_generator/pyproject.toml

```
@@ -1,41 +1,12 @@
 [project]
 name = "railjson_generator"
 version = "0.4.9"
-
-[tool.poetry]
-name = "railjson_generator"
-version = "0.4.9"
 description = ""
-authors = ["OSRD <contact@osrd.fr>"]
-
-[tool.poetry.dependencies]
-osrd-schemas = { path = "../osrd_schemas/", develop = false }
-pytest = "^7.4.3"
-pytest-cov = "^4.1.0"
-python = ">=3.9,<3.12"
-
-[tool.poetry.group.dev.dependencies]
-black = "^24.4.2"
-flake8-formatter-junit-xml = "^0.0.6"
-isort = "^5.12.0"
-pyproject-flake8 = "^6.0.0.post1"
-pytype = { platform = "linux", version = "^2023.10.17" }
-
-[tool.flake8]
-ignore = "W503,E203"
-max-line-length = 120
-exclude = [".venv/"]
-
-[tool.black]
-line-length = 120
-
-[tool.isort]
-profile = "black"
-multi_line_output = 3
+authors = [{ name = "OSRD Contributors", email = "contact@osrd.fr" }]
+requires-python = ">= 3.11"
 
-[tool.pytype]
-inputs = ["railjson_generator"]
+dependencies = ["osrd-schemas @ ../osrd_schemas"]
 
-[build-system]
-requires = ["poetry-core>=1.0.0"]
-build-backend = "poetry.core.masonry.api"
+[project.optional-dependencies]
+check = ["pytest ~= 7.4.3", "pytest-cov ~= 4.1.0"]
+dev = ["pyright ~= 1.1.393", "ruff ~= 0.9.5"]
```

Aren't `pyright` and `ruff` also checking dependencies ?
```suggestion
check = ["pyright ~= 1.1.393", "pytest ~= 7.4.3", "pytest-cov ~= 4.1.0", "ruff ~= 0.9.5"]
```

---

In my mind, one is for tests, the other is for development tools. But I'm fine putting all of them in the same bag. I just followed a convention that was already here.

---

Ah, I would tend to use the group `dev` for dependencies a dev would use but not the ci 
In this instance there aren't any but for example `jupyter` would fit in `dev` more than in `checks`